### PR TITLE
Catch common cases where Directions are misspelled 

### PR DESCRIPTION
**Pull request by Cypher1**
Unfortunately, because Direction is optional, we can't simply catch any case where an unknown direction is used (as the unknown text might actually be a handle name or type etc).

I think we're starting to run into the limits of what PEGjs can do for us in terms of error messages, but we can continue to add 'common sense' error cases like this when they are common and confusing enough.

### CODE REVIEW

File: src/runtime/manifest-ast-types/enums.ts [hunk 1]
```diff
@@ -11,7 +11,17 @@
...
+  'reads',
+  'writes',
+  'reads writes',
+  'hosts',
+  '`consumes',
+  '`provides',
+  'any'
+] as const;
+
+export type Direction = typeof directions[number];
```
Comment (mykmartin) [inline src/runtime/manifest-ast-types/enums.ts:24]: This is a bit confusing.. what does the `export .. as const` and `typeof directions[number]` do? Might be useful to put a comment in.

**Review by mykmartin — APPROVED** [review]
Nice :)

File: src/runtime/manifest-ast-types/enums.ts (hunk 1 above)
Comment (Cypher1) [reply src/runtime/manifest-ast-types/enums.ts:24]: Well there's a set of parts:
1. export does what it normally does, no special effects in this case
2. the list is just a list
3. `as const` turns off type widening (amongst other things https://blog.logrocket.com/const-assertions-are-the-killer-new-typescript-feature-b73451f35802/ or https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions )
4. the `typeof list_var[number]` forms a union from the values of the list

Of these I think 1,2 are well understood.

I'd love some help crafting that into a single comment if necessary.
Is linking to https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions enough?

In general, I'm not sure that we should be explaining language features like these whenever we use them. There's five places I've introduced `as const` already to aid in type checking.

Comment (mykmartin) [reply src/runtime/manifest-ast-types/enums.ts:24]: Ok, I just hadn't encountered the `as const` notation; that's fine. The `typeof` trick seems more obscure and could have a comment (literally what you just said; "forms a union ...").

How does this structure differ from using a standard TS enum? Why not use one?

Comment (Cypher1) [reply src/runtime/manifest-ast-types/enums.ts:24]: I find that these let you include other lists easily (e.g. including other lists with spread operators which [from memory] requires more type magic to do with enums). There's also no key/value distinction which avoids some mess.

**Pull request closed without merging**